vllm/model_executor/parallel_utils/communication_op.py

```diff
@@ -1,20 +1,51 @@
 import torch
 
 from vllm.model_executor.parallel_utils.parallel_state import (
-    get_tensor_model_parallel_world_size,
-    get_tensor_model_parallel_group,
-)
+    get_tensor_model_parallel_world_size, get_tensor_model_parallel_group,
+    get_tensor_model_parallel_rank)
+from vllm.model_executor.parallel_utils.fast_allreduce import FastAllreduce
 
+fa_handle = None
+is_capturing = False
```

Comment:
Can we move all the variables/functions related to the custom all reduce to `fast_allreduce.py`?

Reply:
Sure